src/L6/datasorter.py

```diff
@@ -43,13 +43,20 @@ def set_input_data(self, file_path_name: str):
             raise NotCSVException
         else:
             self.data = []
-            with open(file_path_name, newline= '') as csvfile:
-                dialect = csv.Sniffer().sniff(csvfile.read(1024), [',','|'])
+            with open(file_path_name, newline='') as csvfile:
+                try:
+                    dialect = csv.Sniffer().sniff(csvfile.read(1024), [',', '|'])
+                except csv.Error:
+                    raise NotCSVException
                 csvfile.seek(0)
-                items = csv.reader(csvfile,dialect)
+                items = csv.reader(csvfile, dialect)
                 for row in items:
                     for i in row:
-                        self.data.append(i)
+                        try:
+                            value = float(i)
+                            self.data.append(value)
+                        except ValueError:
```

Comment:
if you are catching a ValueError just to raise another ValueError, there's no point in catching the error in the first place.

A better approach could be to replace the try-except block with a call to `str.isdigit` which returns a simple boolean, like this:
Instead of:
```python
try:
    value = float(i)
except ValueError:
```
Use (assuming `i` is a `str`):
```python
if i.isdigit():
    self.data.append(float(i))
else:
    raise ValueError("The CSV contains non-numeric values")
```